crates/aptos-build-info/src/lib.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::BTreeMap;
+
+use shadow_rs::shadow;
+
+/// Build information keys
+pub const BUILD_BRANCH: &str = "build_branch";
+pub const BUILD_CARGO_VERSION: &str = "build_cargo_version";
+pub const BUILD_COMMIT_HASH: &str = "build_commit_hash";
+pub const BUILD_TAG: &str = "build_tag";
+pub const BUILD_TIME: &str = "build_time";
+pub const BUILD_OS: &str = "build_os";
+pub const BUILD_PKG_VERSION: &str = "build_pkg_version";
+pub const BUILD_RUST_CHANNEL: &str = "build_rust_channel";
+pub const BUILD_RUST_VERSION: &str = "build_rust_version";
+
+pub type BuildInfomation = BTreeMap<String, String>;
+
+pub fn get_build_information() -> BuildInfomation {
+    shadow!(build);
+
+    let mut build_information = BTreeMap::new();
+
+    // Get Git metadata from shadow_rs crate.
+    // This is applicable for native builds where the cargo has
+    // access to the .git directory.
+    build_information.insert(BUILD_BRANCH.into(), build::BRANCH.into());
+    build_information.insert(BUILD_CARGO_VERSION.into(), build::CARGO_VERSION.into());
+    build_information.insert(BUILD_COMMIT_HASH.into(), build::COMMIT_HASH.into());
+    build_information.insert(BUILD_TAG.into(), build::TAG.into());
+    build_information.insert(BUILD_TIME.into(), build::BUILD_TIME.into());
+    build_information.insert(BUILD_OS.into(), build::BUILD_OS.into());
+    build_information.insert(BUILD_PKG_VERSION.into(), build::PKG_VERSION.into());
```

Comment:
Note that this will be the version of the `aptos-build-info` and not the downstream crate (which is why we had a macro before)

Reply:
Good catch. Interesting.

I have used a combination of method calls and macro to make this setup still work. Let me know.